spec/routemaster/integration/api_client_spec.rb

```diff
@@ -297,14 +298,15 @@ def timestamp ; subject.get(url).body.updated_at ; end
 
     it 'traverses through pagination next all links that match the request params' do
       res = subject.discover(url)
+
       expect(res.resources.index(filters: { first_name: 'roo' }).count).to eq(5)
     end
 
-    it 'does not make any http requests to fetch resources any if just the index method is called' do
-      resources = subject.discover(url).resources
+    it 'does not make any http requests to fetch individual resources if just the index method is called' do
```

Comment:
This diff is here just because this test became a flakey after my changes, so I changed it to test the specific behaviour of the `index` method.

I agree that the caching of the root is missing altogether so I'll add a spec for that (i.e. caching the request on redis and avoid going over the wire), but it's probably overkill to test the memoization itself (as in testing if the `get` method is not called inside the `ApiClient` altogether).